test/system/registration_test.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require "application_system_test_case"
+
+class RegistrationFlowTest < ApplicationSystemTestCase
+  test "register user" do
+    fake_origin = ENV['WEBAUTHN_ORIGIN']
+    fake_client = WebAuthn::FakeClient.new(fake_origin)
+
+    fixed_challenge = SecureRandom.random_bytes(32)
+    WebAuthn::CredentialCreationOptions.stub_any_instance :challenge, fixed_challenge do
+      fake_credentials = fake_client.create(challenge: fixed_challenge)
+      register_user(fake_credentials: fake_credentials)
+    end
```

Comment:
I read this a couple of time and I am not sure I clearly can distinguish the separate test case steps, i.e. "Setup", "Actions", "Expectations".

Expectations are clear. But it seems like "Setup" and "Actions" steps are a bit mixed here and inside `register_user` test helper maybe.

Reply:
Logically it would be:

- Given that the create is stubbed with the follwing...
- When the user enters the username and clicks on Register using WebAuthn
- Then the following assertions...

Reply:
In summary, I think we need to make the test case user actions more visible and explicit here.